src/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -1,36 +1,37 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-
-import './style.css';
-
-class ErrorBoundary extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { hasError: false };
-  }
-
-  static getDerivedStateFromError() {
-    return { hasError: true };
-  }
-
-  render() {
-    const { hasError } = this.state;
-    const { children } = this.props;
-
-    if (hasError) {
-      return (
-        <h1 className="error-message">
-          Something went wrong ☹
-        </h1>
-      );
-    }
-
-    return children;
-  }
-}
-
-ErrorBoundary.propTypes = {
-  children: PropTypes.element.isRequired,
-};
-
-export default ErrorBoundary;
+import React, { ReactElement } from 'react';
+
+import './style.css';
+
+type ErrorBoundaryProps = {
+  children: ReactElement
```

Comment:
There is already PropsWithchildren type in react.

Reply:
Changed ReactElement to PropsWithchildren.